### PR TITLE
Generated Microsoft.Resources package with 2018-05-01 api-version

### PR DESCRIPTION
**Pull request by Tiano2017**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] Please add REST spec PR link to the SDK PR
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/Resource/Resource.Tests/Helpers/ManagementGroupsTestUtilities.cs [hunk 1]
```diff
@@ -1,28 +0,0 @@
...
-using System.Net;
-using System.Text;
-using Microsoft.Azure.Management.ResourceManager;
-using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
-using ResourceGroups.Tests;
-
-namespace Resource.Tests.Helpers
-{
-    public static class ManagementGroupsTestUtilities
-    {
```
Comment (dsgouda) [inline src/SDKs/Resource/Resource.Tests/Helpers/ManagementGroupsTestUtilities.cs]: Tests can be skipped instead of deleted if this is temporary

**Review by dsgouda — CHANGES_REQUESTED** [review]
- Please link the related REST spec
- Please bump the version number in csproj to `1.8.0-preview`
- Please create a generate.p1 file similar to [this ](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Compute/Management.Compute/generate.ps1) and regenerate the code.

**Comment by dsgouda** [comment]
@Tiano2017 Please fix failing tests

**Comment by Tiano2017** [comment]
@dsgouda test failures are fixed.

**Comment by dsgouda** [comment]
@Tiano2017 Please address the other comments.

File: src/SDKs/Resource/Resource.Tests/Helpers/ManagementGroupsTestUtilities.cs (hunk 1 above)
Comment (TianoMS) [reply src/SDKs/Resource/Resource.Tests/Helpers/ManagementGroupsTestUtilities.cs]: the related APIs were removed. I think we can remove the test cases and utilities. 

---
In reply to: [193487458](https://github.com/Azure/azure-sdk-for-net/pull/4418#discussion_r193487458) [](ancestors = 193487458)

**Comment by rajshah11** [comment]
PR looks good as far as ManagementGroups is concerned, since it is in a separate namespace and all files and tests should be removed fro here. Thanks!

File: src/SDKs/Resource/Management.ResourceManager/Microsoft.Azure.Management.ResourceManager.csproj [hunk 2]
```diff
@@ -7,7 +7,7 @@
 		<PackageId>Microsoft.Azure.Management.ResourceManager</PackageId>
 		<Description>Provides resource group and resource management capabilities for Microsoft Azure.</Description>
 		<AssemblyName>Microsoft.Azure.Management.ResourceManager</AssemblyName>
-		<Version>1.7.0-preview</Version>
+		<Version>1.8.0-preview</Version>
 		<PackageTags>Microsoft Azure resource management;resource management;resource groups;</PackageTags>
```
Comment (dsgouda) [inline src/SDKs/Resource/Management.ResourceManager/Microsoft.Azure.Management.ResourceManager.csproj]: Please add `PackageReleaseNotes` similar to [this](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Compute/Management.Compute/Microsoft.Azure.Management.Compute.csproj#L13)

**Review by dsgouda — CHANGES_REQUESTED** [review]
@Tiano2017 
- Please run `msbuild build.proj /t:Build /p:Scope=SDKs\Resource` and commit changes to the `.props` file
- Create a generate.ps1 similar to [this](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Compute/Management.Compute/generate.ps1) and regenerate the code


Looks good otherwise

File: src/SDKs/Resource/Management.ResourceManager/Microsoft.Azure.Management.ResourceManager.csproj (hunk 2 above)
Comment (Tiano2017) [reply src/SDKs/Resource/Management.ResourceManager/Microsoft.Azure.Management.ResourceManager.csproj]: @dsgouda I run msbuild build.proj /t:Build /p:Scope=SDKs\Resource but didn't see any change being generated. Did I understand it correctly?
Can you give more details about generate.ps1? today we have generate.cmd: https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Resource/Management.ResourceManager/generate.cmd, is that  not the right way?

Comment (dsgouda) [reply src/SDKs/Resource/Management.ResourceManager/Microsoft.Azure.Management.ResourceManager.csproj]: @Tiano2017 
If msbuild did not generate any artifacts, the PR should be fine
We are moving away from using .cmd script to generate the code. Hence the .ps1 script.

Comment (Tiano2017) [reply src/SDKs/Resource/Management.ResourceManager/Microsoft.Azure.Management.ResourceManager.csproj]: 'Start-AutoRestCodeGeneration' is not recognized. Do I need to install anything?

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
@Tiano2017 can you build the SDK at least once using
msbuild build.proj /t:Build /p:Scope=SDKs\Resource
this will create the much needed meta data that is required to publish the SDK

**Comment by Tiano2017** [comment]
@shahabhijeet I did that but didn't see any change produced.

File: src/SDKs/Resource/Management.ResourceManager/Microsoft.Azure.Management.ResourceManager.csproj (hunk 2 above)
Comment (Tiano2017) [reply src/SDKs/Resource/Management.ResourceManager/Microsoft.Azure.Management.ResourceManager.csproj]: @dsgouda any clue on this?

**Comment by shahabhijeet** [comment]
I cloned your fork and here is what I get when I build your SDK from your fork
Here are the API version that are currently baked in the SDK
Features_2015-12-01;ManagementLinkClient_2016-09-01;Resources_2016-09-01;Authorization_2016-09-01;Authorization_2018-03-01;Management_2018-03-01;PolicyClient_2018-03-01;ResourceManagementClient_2018-05-01;Resources_2018-05-01;SubscriptionClient_2016-06-01

This meta data is essential for the SDK package

File: src/SDKs/Resource/Management.ResourceManager/Microsoft.Azure.Management.ResourceManager.csproj (hunk 2 above)
Comment (dsgouda) [reply src/SDKs/Resource/Management.ResourceManager/Microsoft.Azure.Management.ResourceManager.csproj]: Please pull the latest upstream changes and run `msbuild build.proj` this should install the cmdlet for you

**Review by shahabhijeet — APPROVED** [review]

**Pull request closed without merging**